session/pingpong/invoice_tracker.go

```diff
@@ -359,9 +359,7 @@ func (it *InvoiceTracker) WaitFirstInvoice(wait time.Duration) error {
 				return nil
 			}
 		case <-timeout:
-			// TODO uncomment this error return when all consumers will be able to pay for the first invoice before session start.
-			// return fmt.Errorf("failed waiting for first invoice")
-			return nil
+			return fmt.Errorf("failed waiting for first invoice")
```

Comment:
In my provider logs I see clear 1st invoice failure, but it never triggers `Payment engine error` flow.
```
:455  > Marking invoice as not sent error="timeout waiting for reply to \"p2p-payment-invoice\": p2p send timeout
```

Checked and reason is that this error is being swallowed here:
<img width="728" alt="Screen Shot 2020-06-18 at 15 03 51" src="https://user-images.githubusercontent.com/2713911/85018984-bb9b5d00-b176-11ea-9f96-e2512dc7f09f.png">

As we don't do retries in `sendFirstInvoice()` this strange swallowing condition should only where `sendInvoice()` is used within retry loop.

Reply:
Added retries for sending the first invoice.